### ISSUE TITLE
Log details for appointments with "PID" in the comment field

### ISSUE THREAD

**Issue by cferris32** [department-of-veterans-affairs/va.gov-team#60084]
This work is to track info around appointments that have "PID" in the comment field, to address user feedback wondering what PID means. This is a continuation of the logging efforts from this ticket, and will follow the same framework: https://app.zenhub.com/workspaces/appointments-team-603fdef281af6500110a1691/issues/gh/department-of-veterans-affairs/va.gov-team/58400

AC:
[ ] - logging is implemented to capture appropriate details around appointments with a "PID" comment

**Comment by cferris32** [comment]
Logging findings:

After a few days of gathering logs we have already seen a couple hundred appointments come through with "PID" in the comments. Most appear to be of administrative nature, possibly an admin comment left by a scheduler helping facilitate the appointment. Here is a count snapshot of the first few days:
![Screenshot 2023-07-12 at 3.03.35 PM.png](https://images.zenhubusercontent.com/5fcfbc1c239ebd98b2acfe6c/124c08bf-47a2-44a1-930b-d38d9eebf56b)

Here are the different comments found containing the substring "PID", most appear to be admin notes:
![Screenshot 2023-07-12 at 3.11.02 PM.png](https://images.zenhubusercontent.com/5fcfbc1c239ebd98b2acfe6c/790ccb54-0faf-4d6b-83fc-3018ada5ff32)

The clinics associated with these appointments have mostly been null, with a few exceptions:
![Screenshot 2023-07-12 at 3.11.17 PM.png](https://images.zenhubusercontent.com/5fcfbc1c239ebd98b2acfe6c/a013c721-3c9c-4ede-bad6-d2d0494127a7)

All logged appointments so far have come from calls to fetch appointments:
![Screenshot 2023-07-12 at 3.12.22 PM.png](https://images.zenhubusercontent.com/5fcfbc1c239ebd98b2acfe6c/5a11789b-ce44-4b0c-b61f-fde86ec3944d)

There has also been a fairly wide distribution of location ids associated with these appointments:
![Screenshot 2023-07-12 at 3.12.51 PM.png](https://images.zenhubusercontent.com/5fcfbc1c239ebd98b2acfe6c/761653b6-9294-491a-afc3-1e0681af30e9)

Finally, we had a few logs come through containing more uniquely formatted comments with fuller instructions, seeming to be admin notes related to rescheduling the appointment:
![Screenshot 2023-07-12 at 3.10.05 PM.png](https://images.zenhubusercontent.com/5fcfbc1c239ebd98b2acfe6c/4ac87a81-af45-4c7e-a28f-478c8da7aed6)

In summary it appears that we have a notable number of appointments containing "PID" in the comments, and these comments are usually containing notes of administrative nature. There is a fair distribution of what location ids are associated with these appointments as well. This information will hopefully inform next steps of research and discovery.

**Comment by cferris32** [comment]
Appointment endpoint key:

Appt Index = GET '/vaos/v1/patients/<icn>/appointments'
Appt Show = GET '/vaos/v1/patients/<icn>/appointments/<id>'
Appt Create = POST '/vaos/v1/patients/<icn>/appointments'

**Comment by cferris32** [comment]
New round of reports for the week of 7/28:

Over this week, we've had a little under 300 new logs come in with "PID" comments. We are now also capturing the appointment ids, with the range shown in the log group below:
![Screenshot 2023-07-27 at 3.53.22 PM.png](https://images.zenhubusercontent.com/5fcfbc1c239ebd98b2acfe6c/1ab39ce6-2bed-4460-b7d3-16f63e770a84)

The majority of our clinic ids remain null, with a handful of outliers:
![Screenshot 2023-07-27 at 3.53.42 PM.png](https://images.zenhubusercontent.com/5fcfbc1c239ebd98b2acfe6c/e39a11d7-c514-47c8-8528-01454ce2d74b)

Most of our "PID" content has come from the comment field, but a few have now been coming through reason_code too:
![Screenshot 2023-07-27 at 3.55.51 PM.png](https://images.zenhubusercontent.com/5fcfbc1c239ebd98b2acfe6c/d756a2fc-fb9a-4fb1-a566-6d4fc27ff72f)

Historically, all of our PID logs have come from fetching appointments, but we've seen one new log this week from an appointment creation:
![Screenshot 2023-07-27 at 3.56.23 PM.png](https://images.zenhubusercontent.com/5fcfbc1c239ebd98b2acfe6c/b1e75d07-7f91-46b6-9cd1-5aa79d865146)

In terms of our endpoints, the majority have been on fetching the appointment list, with one of those 266 being the aforementioned POST request. We've seen a handful come through for fetching individual appointments.
![Screenshot 2023-07-27 at 3.56.34 PM.png](https://images.zenhubusercontent.com/5fcfbc1c239ebd98b2acfe6c/91a5a119-1ca9-4c71-b1ca-f6de55b672ac)

Here is the log contents for the new POST request, being the only one of its kind thus far:
![Screenshot 2023-07-27 at 4.17.10 PM.png](https://images.zenhubusercontent.com/5fcfbc1c239ebd98b2acfe6c/09c69b08-b990-4afc-a131-948e4935e6ab)

Here are the comment contents for the newly gathered logs:
![Screenshot 2023-07-27 at 4.17.41 PM.png](https://images.zenhubusercontent.com/5fcfbc1c239ebd98b2acfe6c/b8cc1898-c885-4e88-b666-4b3eb85dbe07)

**Comment by cferris32** [comment]
Reports for first week of August:

We're now returning appointment status in our logs, which reveals the following trends in statuses among our PID appointments:
![Screenshot 2023-08-08 at 3.45.33 PM.png](https://images.zenhubusercontent.com/5fcfbc1c239ebd98b2acfe6c/2513a9c4-7415-48d6-893b-73cfa0408501)

Here's a snapshot of the top comment pattern groups from this round of logs:
![Screenshot 2023-08-08 at 3.46.27 PM.png](https://images.zenhubusercontent.com/5fcfbc1c239ebd98b2acfe6c/7bc9883c-384e-462d-bfb5-86f27ad1c881)

Zooming in on one example from this snapshot, the "PER TEXT ORDER" comments in full look like this:
![Screenshot 2023-08-09 at 11.55.48 AM.png](https://images.zenhubusercontent.com/5fcfbc1c239ebd98b2acfe6c/430453d5-1c92-4640-9738-77c4449e26e9)

From this same dataset, we can see that multiple entries are of the same appointment id being fetched:
![Screenshot 2023-08-09 at 12.12.47 PM.png](https://images.zenhubusercontent.com/5fcfbc1c239ebd98b2acfe6c/86dda127-ac53-4b3e-8de6-44354b1dfe79)

And this aligns with previous trends we've seen, and continue to see, where the vast majority of these appointments are coming from fetches:
![Screenshot 2023-08-09 at 12.13.30 PM.png](https://images.zenhubusercontent.com/5fcfbc1c239ebd98b2acfe6c/4fbe1d1d-2dc1-432e-b8ea-48fcc62c5ef5)

From these iterations we've begun to notice a few key overall trends - these PID logs are mostly from users retrieving their booked appointments, with the "PID" section of the comment most often being in the form of "PID mm/dd/yy". We may gain new insight from determining what "PID" means in this particular context.

**Comment by ldelacosta** [comment]
@JRRoof - friendly reminder to send the spreadsheet to Kay. 

@young1205 - I would keep this open until we are done with tracking the data in Loki. If there is nothing else we need to do, we can close it out. Best to check with Jeff, @simiadebowale @cferris32 for the latest update. Thanks!

**Comment by ldelacosta** [comment]
@cferris32 - could we get the latest data from this tracking? Once we have that available, we can stop the logging. You can attach the results in the ticket. Thanks!

**Comment by cferris32** [comment]
Most recent logging report from the last two weeks, 2/20 - 3/5

We've received nearly 30k instances of "PID" comments in this timeframe:
<img width="614" src="https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBMVM3QXc9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--f0d6433d2bd5d80c65f10b7e30f47de9abe7deb8/Screenshot%202024-03-05%20at%204.03.03%20PM.png" alt="Screenshot 2024-03-05 at 4.03.03 PM.png" />

The proportion of appointment statuses within this sample are as such:
<img width="1756" src="https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBM0s3QXc9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--bc7b2d6136696bae9a7247468da5cafa99caa398/Screenshot%202024-03-05%20at%204.03.38%20PM.png" alt="Screenshot 2024-03-05 at 4.03.38 PM.png" />

Clinic numbers were mostly absent from this sample, although some exceptions can be found:
<img width="1752" src="https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBM2U3QXc9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--2a9e0e15f2df1994306168fd0d7cfcdcafb33bd0/Screenshot%202024-03-05%20at%204.05.33%20PM.png" alt="Screenshot 2024-03-05 at 4.05.33 PM.png" />

Digging into the appointment for clinic 22624 revealed the following details:
<img width="2000" src="https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBM3E3QXc9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--1c3bf056b6700dcf26ad4ce398ee053409e24e90/Screenshot%202024-03-05%20at%204.31.53%20PM.png" alt="Screenshot 2024-03-05 at 4.31.53 PM.png" />

Here are the most frequent values for location ids from the sample:
<img width="1598" src="https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBM3U3QXc9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--3ab9c068eb57951167095db998204836f8d20b8d/Screenshot%202024-03-05%20at%204.05.59%20PM.png" alt="Screenshot 2024-03-05 at 4.05.59 PM.png" />

The most common comments containing "PID":
<img width="1620" src="https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBMzI3QXc9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--648abb848552a62d2664d3322a6f4a8368d57ab9/Screenshot%202024-03-05%20at%204.06.08%20PM.png" alt="Screenshot 2024-03-05 at 4.06.08 PM.png" />

We can also see the ratio of how many of these substrings came through the comment field vs the reason code field:
<img width="1754" src="https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBNEM3QXc9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--5ae8c5c2674483f365928884e466afc2d09402db/Screenshot%202024-03-05%20at%204.06.21%20PM.png" alt="Screenshot 2024-03-05 at 4.06.21 PM.png" />

Some other contexts in which "PID" occurred include using it to refer to "pelvic inflammatory disease":
<img width="1722" src="https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBNUM3QXc9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--d694522d73914732de44c22ed1fc7345c5566cb2/Screenshot%202024-03-06%20at%202.05.36%20PM.png" alt="Screenshot 2024-03-06 at 2.05.36 PM.png" />

This concludes the logging report for 2/20-3/5.

**Comment by ldelacosta** [comment]
If you could turn off the logging and will let you know if we do need to turn it on again. Let me know when it's done. @cferris32

**Comment by ldelacosta** [comment]
@cferris32 - please confirm in the ticket when the logging has stopped. Thanks!

**Comment by cferris32** [comment]
Logging for PID information has been turned off. Closing out this ticket.
